src/store/metrics.js

```diff
@@ -0,0 +1,224 @@
+import Vue from 'vue'
+import store from '@/store/index'
+import { MetricHelper } from '@/metric'
+
+export default {
+  namespaced: true,
+  state: {
+    metrics: {}
+  },
+  getters: {
+    getMetricKey: (state) => (metricName) => {
+      const metricEntry = Object.entries(state.metrics).find(([_, metric]) => metric.name === metricName)
+      return metricEntry === undefined ? undefined : metricEntry[0]
+    },
+    getMetricDrawState: (state) => (metricName) => {
+      const metric = Object.entries(state.metrics).find(([_, metric]) => metric.name === metricName)[1]
+      return {
+        drawMin: metric.drawMin,
+        drawAvg: metric.drawAvg,
+        drawMax: metric.drawMax
+      }
+    },
+    getAllMetricKeys: (state) => () => {
+      return Object.keys(state.metrics)
+    },
+    getAllMetrics: (state) => () => {
+      return Object.values(state.metrics)
+    }
+  },
+  mutations: {
+    resetAllMetrics (state) {
+      console.log('Reset allMetrics')
+      Object.keys(state.metrics).forEach(function (key) { if (!key.startsWith('_')) { Vue.delete(state.metrics, key) } })
+    },
+    removeMetricEntry (state, { metricKey }) {
```

Comment:
just... `deleteMetric`?

Reply:
There is already an action `deleteMetric` which calls `removeMetricEntry` and `checkMetricDrawState`